### DISCUSSION TITLE
[New] Économiser de la bande passante grace à un ServiceWorker

### DISCUSSION THREAD

**Discussion by tbroyer** [cnumr/best-practices#120] [Bonnes pratiques] [answered]
Copie de https://groups.google.com/g/ecoconceptionweb/c/1zZ8dqSCXoc/m/m2yUCD8GCAAJ

> https://blog.atolcd.com/best-of-tech-15/#des-pages-instantanees-et-des-teraoctets-economises-web-service-worker
> 
> Pour coller au max à l'écoconception, on évitera peut-être de précharger l'entête et le footer lors de l'installation du SW pour privilégier un chargement "lazy" lors de la navigation suivante. Et on n'oubliera pas d'utiliser HTTP/2 si possible pour améliorer encore les choses (multiplexing sur la même connexion TCP, entêtes HTTP compressées et dans un format "binaire") lors du chargement de ces 3 parties de la page (entête, contenu, footer), et bien sûr des ressources associées (images, etc.).

Je ne sais pas comment on pourrait tourner cette BP, mais ça me semble une approche intéressante (proche du PJAX mais sans ses inconvénients; voir https://youtu.be/ivLhf3hq7eM?t=423 par exemple)

On pourrait par contre assez facilement ajouter une BP sur le **mauvais** usage de Service Worker (surtout de la mise en cache proactive). Peut-être orienté "offline-first", où ça ne devrait peut-être pas être l'approche par défaut (par exemple https://hpbn.co/ qui se met en cache dès la première visite, sans rien demander)

**Comment by ACTLEM** [comment, marked as answer]
Note équipe projet: Ok pour la création de cette bonne pratique (discussion en cours sur une liée: https://github.com/cnumr/best-practices/discussions/122)

Issue créée: https://github.com/cnumr/best-practices/issues/171
